rules/internal/utils/http.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"github.com/golang/protobuf/proto"
+	"github.com/jhump/protoreflect/desc"
+	"google.golang.org/genproto/googleapis/api/annotations"
+)
+
+// GetHTTPRules returns a slice of HTTP rules for a given method descriptor.
+//
+// Note: This returns a slice -- it takes the google.api.http annotation,
+// and then flattens the values in `additional_bindings`.
+// This allows rule authors to simply range over all of the HTTP rules,
+// since the common case is to want to apply the checks to all of them.
+func GetHTTPRules(m *desc.MethodDescriptor) []*annotations.HttpRule {
+	rules := []*annotations.HttpRule{}
+
+	// Get the method options.
+	opts := m.GetMethodOptions()
+
+	// Get the "primary" rule (the direct google.api.http annotation).
+	var httpRule *annotations.HttpRule
+	if x, err := proto.GetExtension(opts, annotations.E_Http); err == nil {
+		httpRule = x.(*annotations.HttpRule)
+		rules = append(rules, httpRule)
+	}
+
+	// Iterate over any additional bindings and flatten them into the
+	// `rules` array.
+	for _, binding := range httpRule.GetAdditionalBindings() {
```

Comment:
This looks like it risks a nil pointer dereference if err on line 37 != nil

Reply:
Nevermind, they have [defensive checks](https://github.com/googleapis/go-genproto/blob/master/googleapis/api/annotations/http.pb.go#L527-L532) in place